Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -764,38 +767,42 @@ public class GraphTraverser: GraphTraversing {
                 bcsymbolmapPaths: bcsymbolmapPaths,
                 linking: linking,
                 architectures: architectures,
-                product: (linking == .static) ? .staticFramework : .framework
+                product: (linking == .static) ? .staticFramework : .framework,
+                platformFilters: platformFilters
             )
         case let .library(path, _, linking, architectures, _):
             return .library(
                 path: path,
                 linking: linking,
                 architectures: architectures,
-                product: (linking == .static) ? .staticLibrary : .dynamicLibrary
+                product: (linking == .static) ? .staticLibrary : .dynamicLibrary,
+                platformFilters: platformFilters
             )
         case let .bundle(path):
-            return .bundle(path: path)
+            return .bundle(path: path, platformFilters: platformFilters)
         case .packageProduct:
             return nil
         case let .sdk(_, path, status, source):
             return .sdk(
                 path: path,
                 status: status,
-                source: source
+                source: source,
+                platformFilters: platformFilters
             )
         case let .target(name, path):
             guard let target = target(path: path, name: name) else { return nil }
             return .product(
                 target: target.target.name,
                 productName: target.target.productNameWithExtension,
-                platformFilters: target.target.dependencyPlatformFilters
+                platformFilters: target.target.dependencyPlatformFilters // What to do here now that we have smart things?
```

Comment:
we have a few options:

- We can either omit `platformFilters` from `product` dependencies (i.e. `.target` / `.project`) dependencies and automatically derive this as we do today on `main`
- if we expose `platformFilters` we'd have to do some form of intersection between the user specified filters (which may contain errors - possibly lint warn / error ahead of time) + valid filters we determine